### PR TITLE
Azure labels and tag mapping support for a new refresh

### PR DESCRIPTION
**Pull request by AlexanderZagaynov**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1493041
related old refresh type PR: https://github.com/ManageIQ/manageiq-providers-azure/pull/231
related core repo PR: https://github.com/ManageIQ/manageiq/pull/17212
related UI repo PR: https://github.com/ManageIQ/manageiq-ui-classic/pull/3697

### CODE REVIEW

**Comment by bronaghs** [comment]
@AlexanderZagaynov - the BZ for this PR has a target release of 5.8.4 (Fine branch) and this PR is specific to the Graph refresh which is not supported in the Fine release, so this PR cannot be backported. You will have to make a separate PR for the Fine branch to add tagging support for the old refresh. Getting this into the Fine branch is the priority.

File: app/models/manageiq/providers/azure/inventory_collection_default/cloud_manager.rb [hunk 1]
```diff
@@ -60,6 +60,54 @@ def key_pairs(extra_attributes = {})
       super(attributes.merge!(extra_attributes))
     end
 
+    def vm_and_template_labels(extra_attributes = {})
```
Comment (cben) [inline app/models/manageiq/providers/azure/inventory_collection_default/cloud_manager.rb]: @Ladas @juliancheal  these collections are starting to pile up in multiple providers, do you see a way to share them?
(I'm fine with duplicating for gaprindashvili & fine, then refactoring)

Comment (cben) [reply app/models/manageiq/providers/azure/inventory_collection_default/cloud_manager.rb]: to clarify, I'm not sure sharing vm_and_template_labels/taggings between 2 providers is a big win;
I'm more thinking of a "polymorphic" anything-custom-attributes, anything-taggings, something like these:
https://github.com/ManageIQ/manageiq-providers-kubernetes/blob/5ec31943f8/app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb#L359-L400

Comment (Ladas) [reply app/models/manageiq/providers/azure/inventory_collection_default/cloud_manager.rb]: yeah, we should probably try to unify these next, from what I saw, each provider had it slightly different though

**Comment by JPrause** [comment]
@miq-bot add_label blocker

**Comment by JPrause** [comment]
@AlexanderZagaynov do you have an update on the status of this PR.

File: app/models/manageiq/providers/azure/inventory_collection_default/cloud_manager.rb (hunk 1 above)
Comment (juliancheal) [reply app/models/manageiq/providers/azure/inventory_collection_default/cloud_manager.rb]: Yeah sounds good to me.

**Comment by cben** [comment]
Looks great :+1:

I don't remember what you decided about 'Vm' type - do you need separate 'AzureVm', 'AmazonVm' types?  IMHO better to get just Vm working & merged first, leave sub-types to a follow-up if desired.

**Comment by bronaghs** [comment]
hi @AlexanderZagaynov - Graph refresh was not backported to the Fine branch, so this PR cannot be backported to Fine either. I see you closed the Fine specific implementation https://github.com/ManageIQ/manageiq-providers-azure/pull/231

**Comment by AlexanderZagaynov** [comment]
https://travis-ci.org/ManageIQ/manageiq-providers-azure/jobs/364111457#L1958 failing because of https://github.com/ManageIQ/manageiq/pull/17212 not merged yet

**Comment by JPrause** [comment]
@AlexanderZagaynov can you add the gaprindashvili/yes and the fine/yes if this can be backported.

**Comment by AlexanderZagaynov** [comment]
@miq-bot add_label gaprindashvili/yes
@miq-bot add_label fine/no

File: app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb [hunk 2]
```diff
@@ -279,6 +283,8 @@ def parse_instance(instance)
           :orchestration_stack => @data_index.fetch_path(:orchestration_stacks, @resource_to_stack[uid]),
           :availability_zone   => @data_index.fetch_path(:availability_zones, 'default'),
           :resource_group      => @data_index.fetch_path(:resource_groups, rg_ems_ref),
+          :labels              => labels,
+          :tags                => map_labels('Vm', labels),
```
Comment (cben) [inline app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb]: FYI I think this should be 'VmAzure' like in #231, this is based on top of older version of BZ-1493041-old-refresh branch...

**Comment by AlexanderZagaynov** [comment]
@JPrause there will be a special version of https://github.com/ManageIQ/manageiq-providers-azure/pull/231 for a `fine` branch

File: app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb (hunk 2 above)
Comment (AlexanderZagaynov) [reply app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb]: I'm going to rebase and add some changes to this PR after https://github.com/ManageIQ/manageiq-providers-azure/pull/231 will be merged

**Comment by AlexanderZagaynov** [comment]
@miq-bot remove-label unmergeable

**Comment by AlexanderZagaynov** [comment]
@cben @Ladas @bronaghs please review

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 769aee2739e4dd00c5aea9f88b63b4373a33faa9
Author: Ladislav Smola <lsmola@redhat.com>
Date:   Thu Apr 19 17:58:45 2018 +0200

    Merge pull request #229 from AlexanderZagaynov/BZ-1493041
    
    Azure labels and tag mapping support for a new refresh
    (cherry picked from commit a1e134c12beaf61921e9e59bad97b0bcf28eeaa9)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1568807
```

**Pull request merged**